src/js/oncoprintshapetosvg.js

```diff
@@ -24,20 +24,29 @@ var extractColor = function(str) {
     };
 };
 
+var makeSVGRectBackground = function () {
+    return makeSVGElement('rect', {width: 6, height: 20, x: 0, y: 0, fill: extractColor('rgba(190, 190, 190, 1)').rgb,
+        'fill-opacity': 1
+    });
+};
+
 var rectangleToSVG = function (params, offset_x, offset_y) {
     var stroke_color = extractColor(params.stroke);
     var fill_color = extractColor(params.fill);
-    return makeSVGElement('rect', {
-	width: params.width,
-	height: params.height,
-	x: parseFloat(params.x) + offset_x,
-	y: parseFloat(params.y) + offset_y,
-	stroke: stroke_color.rgb,
-	'stroke-opacity': stroke_color.opacity,
-	'stroke-width': params['stroke-width'],
-	fill: fill_color.rgb,
-	'fill-opacity': fill_color.opacity
-    });
+    var el = document.createElementNS('http://www.w3.org/2000/svg', 'g');
+    el.appendChild(makeSVGRectBackground()); // add grey background
```

Comment:
this feels hacky to me - to generate the rectangle background we should attach an extra rectangle Shape model to be rendered, instead of corrupting rectangleToSVG for a use-specific purpose

Reply:
instead we should add something to `Rule` like `baseLegendShapes` or something. Just as you can see the use of the member `target` here: https://github.com/cBioPortal/oncoprintjs/blob/master/src/js/oncoprintlegendrenderer.js#L122 , which is something attached to the rule for the sole purpose of legend rendering

Reply:
Which target in the code are you reffering to?

Reply:
oops, looks like the file line number changed - now its here: https://github.com/cBioPortal/oncoprintjs/blob/master/src/js/oncoprintlegendrenderer.js#L156

`target` is a member of `rule.legend_config` that, if the rule is so configured, will be the data that the rule is applied to to generate the shape in the legend. 

It has basically nothing to do with this task except that its something attached to the rule thats used for the legend. So my idea is to add something like `base_shapes` to `legend_config`, and we can specify the gray rectangles there.